AmplifyPlugins/Core/AWSPluginsCore/ServiceConfiguration/AmplifyAWSServiceConfiguration.swift

```diff
@@ -8,18 +8,78 @@
 import Foundation
 import AWSClientRuntime
 
+#if canImport(UIKit)
+import UIKit
+#endif
+
+/// Convenience class that is used by Amplify to include metadata such as values for a "User-Agent" during
+/// server interactions.
+///
+/// - Tag: AmplifyAWSServiceConfiguration
 public class AmplifyAWSServiceConfiguration {
 
+    /// - Tag: AmplifyAWSServiceConfiguration.amplifyVersion
     public static let amplifyVersion = "2.10.0"
+
+    /// - Tag: AmplifyAWSServiceConfiguration.platformName
     public static let platformName = "amplify-swift"
 
+    /// Returns basic amount of metadata that includes both
+    /// [AmplifyAWSServiceConfiguration.amplifyVersion](x-source-tag://AmplifyAWSServiceConfiguration.amplifyVersion)
+    /// and
+    /// [AmplifyAWSServiceConfiguration.platformName](x-source-tag://AmplifyAWSServiceConfiguration.platformName)
+    ///
+    /// - Tag: AmplifyAWSServiceConfiguration.frameworkMetaData
     public static func frameworkMetaData() -> FrameworkMetadata {
+        return frameworkMetaData(includeOS: false)
+    }
+
+    /// Returns basic amount of metadata that includes both
+    /// [AmplifyAWSServiceConfiguration.amplifyVersion](x-source-tag://AmplifyAWSServiceConfiguration.amplifyVersion)
+    /// and
+    /// [AmplifyAWSServiceConfiguration.platformName](x-source-tag://AmplifyAWSServiceConfiguration.platformName)
+    /// in addition to the operating system version.
+    ///
+    /// - Tag: AmplifyAWSServiceConfiguration.frameworkMetaDataWithOS
+    public static func frameworkMetaDataWithOS() -> FrameworkMetadata {
+        return frameworkMetaData(includeOS: true)
+    }
 
+    private static func frameworkMetaData(includeOS: Bool) -> FrameworkMetadata {
+        let osKey = "os"
         guard let flutterVersion = platformMapping[Platform.flutter] else {
+            if includeOS {
+                return FrameworkMetadata(
+                    name: platformName,
+                    version: amplifyVersion,
+                    extras: [osKey: frameworkOS()]
+                )
+            }
             return FrameworkMetadata(name: platformName, version: amplifyVersion)
         }
+        var extras = [platformName: amplifyVersion]
+        if includeOS {
+            extras[osKey] = frameworkOS()
+        }
         return FrameworkMetadata(name: Platform.flutter.rawValue,
                                  version: flutterVersion,
-                                 extras: [platformName: amplifyVersion])
+                                 extras: extras)
+    }
+
+    private static func frameworkOS() -> String {
+        // Please note that because the value returned by this function will be
+        // sanitized by FrameworkMetadata by removing anything not in a special
+        // character set that does NOT include the forward slash (/), the
+        // backslash (\) is used as a separator instead.
+        let separator = "\\"
+#if canImport(UIKit)
```

Comment:
We already have a cross-platform [DeviceInfo](https://github.com/aws-amplify/amplify-swift/blob/main/Amplify/Core/Support/DeviceInfo.swift) utility, maybe we can use that one instead?

Reply:
Oh, I missed that - thank you, I'll use that!